test/cloudformation/iam_cloudformation.yaml

```diff
@@ -76,16 +76,24 @@ Resources:
           - Effect: Allow
             Action: iam:PassRole
             Resource: !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/eksctl-*"
+          - Effect: Allow
+            Action:
+              - iam:ListInstanceProfiles
+              - iam:ListInstanceProfileTags
+            Resource:
+              - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:instance-profile/*" 
           - Effect: Allow
             Action:
               - iam:AddRoleToInstanceProfile
               - iam:CreateInstanceProfile
+              - iam:TagInstanceProfile
               - iam:RemoveRoleFromInstanceProfile
               - iam:DeleteInstanceProfile
               - iam:GetInstanceProfile
             Resource:
               - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:instance-profile/KarpenterNodeInstanceProfile-*"
               - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:role/KarpenterNodeRole-*"
+              - !Sub "arn:${AWS::Partition}:iam::${AWS::AccountId}:instance-profile/instance-profile-drift-*"
```

Comment:
I wonder if we should name this to be something like `KarpenterNodeInstanceProfile-Drift-<ClusterName>`